.github/actions/npm-publish-package/action.yml

```diff
@@ -29,16 +29,38 @@ runs:
   steps:
     - uses: actions/setup-node@v3
       with:
-        node-version: 17
+        node-version: 18
+
+    - uses: pnpm/action-setup@v2
+      with:
+        version: 7
+
+    - name: Install action dependencies
+      if: ${{ inputs.release_type == 'beta' }}

```

Comment:
`beta` workflow since completely different from `stable` workflow, should it be in separate actions ? (`npm-publish-package-beta` / `npm-publish-package-stable`)